db/dataset_eval.py

```
@@ -0,0 +1,152 @@
+from db import create_cursor, commit
+import db.exceptions
+import db.dataset
+import db.data
+import jsonschema
+import json
+
+# Job statuses are defined in `eval_job_status` type. See schema definition.
+STATUS_PENDING = "pending"
+STATUS_RUNNING = "running"
+STATUS_DONE = "done"
+STATUS_FAILED = "failed"
+
+
+def evaluate_dataset(dataset_id):
+    """Add dataset into evaluation queue.
+
+    Args:
+        dataset_id: ID of the dataset that needs to be added into the list of
+            jobs.
+
+    Returns:
+        ID of the newly created evaluation job.
+    """
+    with create_cursor() as cursor:
+        cursor.execute(
+            "SELECT count(*) FROM dataset_eval_jobs WHERE dataset_id = %s AND status IN %s",
+            (dataset_id, (STATUS_PENDING, STATUS_RUNNING))
+        )
+        if cursor.fetchone()[0] > 0:
+            raise JobExistsException
+    validate_dataset(db.dataset.get(dataset_id))
```

If this fails we should report the error to the user so that they can fix it